### DISCUSSION TITLE
Detecting YAML format on Ansible repositories

### DISCUSSION THREAD

**Discussion by julien-langlois** [github-linguist/linguist#5165] [Classification] [answered]
## Problem Description

Since most repositories related to Ansible are mostly composed of YAML files, why is this not taken into account for language detection? 
It is often displayed HTML or Shell depending on the files present that are necessary for the roles or playbooks to work properly.

### URL of the affected repository:
https://github.com/geerlingguy/ansible-role-nginx

### Last modified on:
2020-12-07

### Expected language:
YAML

### Detected language:
Either HTML or Shell most of the time

**Comment by lildude** [comment, marked as answer]
> Since most repositories related to Ansible are mostly composed of YAML files, why is this not taken into account for language detection?

Because YAML isn't considered a programming language by Linguist (note: `type: data`) and as such is ignored when calculating the statistics:

https://github.com/github/linguist/blob/633d196c170e002bcccb4143d6906787f360e08b/lib/linguist/languages.yml#L6357-L6383

If you really want your repo to count the YAML files, you can implement the [detectable](https://github.com/github/linguist#detectable) override.

**Reply by cassilup** [reply]
Interesting. Why wouldn't YAML be considered a programming language?
I'm asking in the context of [SCC considering it one](https://github.com/boyter/scc/blob/cb04a8d38a086f4bebef615c4278d131026bef1e/LANGUAGES.md).

**Reply by Alhadis** [reply]
@cassilup The list you're linking to is named _"languages"_,; nowhere is it mentioned that its contents are strictly _programming_ languages. In fact, there are even entries that aren't languages at all:

~~~
License (license,licence,copying,copying3,unlicense,unlicence,license-mit,licence-mit,copyright)
[…]
Systemd (automount,device,link,mount,path,scope,service,slice,socket,swap,target,timer)
~~~
